src/pages/ErrorPages.tsx

```diff
@@ -1,3 +1,133 @@
-export default function ErrorPages(): JSX.Element {
-  return <div>ErrorPages</div>;
+import { keyframes } from '@emotion/react';
+import ErrorOutlineIcon from '@mui/icons-material/ErrorOutline';
+import LockPersonIcon from '@mui/icons-material/LockPerson';
+import ReportProblemIcon from '@mui/icons-material/ReportProblem';
+import RunningWithErrorsIcon from '@mui/icons-material/RunningWithErrors';
+import { Box, Button, styled } from '@mui/joy';
+import CssBaseline from '@mui/joy/CssBaseline';
+import Typography from '@mui/joy/Typography';
+import { useNavigate } from 'react-router-dom';
+
+type ErrorCode = 400 | 403 | 404 | 500;
+interface ErrorPageProps {
+  errorCode?: ErrorCode;
 }
+
+export default function ErrorPages({ errorCode }: ErrorPageProps) {
+  const navigate = useNavigate();
+
+  const redirect = () => navigate(-1);
+
+  const displayError = (errorCode?: ErrorCode) => {
+    switch (errorCode) {
+      case 400:
+        return {
+          status: '400',
+          title: 'Bad Request',
+          subTitle: 'The server could not understand the request due to invalid syntax.',
+          icon: <AnimatedErrorOutlineIcon />,
+        };
+
+      case 403:
+        return {
+          status: '403',
+          title: 'Forbidden',
+          subTitle: 'Sorry, you are not authorized to access this page.',
+          icon: <AnimatedLockPersonIcon />,
+        };
+
+      case 404:
+        return {
+          status: '404',
+          title: 'Not Found',
+          subTitle: 'Sorry, the page you visited does not exist.',
+          icon: <AnimatedReportProblemIcon />,
+        };
+
+      default:
+        return {
+          status: '500',
+          title: 'Internal Server Error',
+          subTitle: 'Sorry, something went wrong.',
+          icon: <AnimatedRunningWithErrorsIcon />,
+        };
+    }
+  };
+
+  const { status, title, subTitle, icon } = displayError(errorCode);
+
+  return (
+    <>
+      <CssBaseline />
+      <Box
+        sx={{
+          display: 'flex',
+          flexDirection: 'column',
+          alignItems: 'center',
+          justifyContent: 'center',
+          height: '80vh',
+          textAlign: 'center',
+        }}
+      >
```

Comment:
Instead of `Box` use `Stack`.

Refs:
- https://mui.com/material-ui/react-box/#introduction
- https://mui.com/material-ui/react-stack/#introduction